addons/mail/static/src/components/attachment_delete_confirm_dialog/attachment_delete_confirm_dialog.js

```diff
@@ -73,6 +73,7 @@ class AttachmentDeleteConfirmDialog extends Component {
         this._dialogRef.comp._close();
         this.attachment.remove();
         this.trigger('o-attachment-removed', { attachmentLocalId: this.props.attachmentLocalId });
+        this.trigger('o-attachments-changed', { hasRecordReloadOnAttachmentsChanged: true });
```

Comment:
You don't need to give the param here (it's read from the chatter itself when `o_attachments_changed` is bound).